tests/components/dlna_dmr/test_media_player.py

```diff
@@ -74,21 +77,47 @@ async def mock_entity_id(
     """
     entity_id = await setup_mock_component(hass, config_entry_mock)
 
+    # Check the entity has registered all needed listeners
+    assert len(config_entry_mock.update_listeners) == 1
+    assert domain_data_mock.async_get_event_notifier.await_count == 1
+    assert domain_data_mock.async_release_event_notifier.await_count == 0
+    assert ssdp_scanner_mock.async_register_callback.await_count == 2
+    assert ssdp_scanner_mock.async_register_callback.return_value.call_count == 0
     assert dmr_device_mock.async_subscribe_services.await_count == 1
+    assert dmr_device_mock.async_unsubscribe_services.await_count == 0
+    assert dmr_device_mock.on_event is not None
 
+    # Run the test
     yield entity_id
 
     # Unload config entry to clean up
     assert await hass.config_entries.async_remove(config_entry_mock.entry_id) == {
```

Comment:
Why do we need to clean up here? We now test this in `test_resource_lifecycle` so we don't need to test it again here if it's the same resources used.

Reply:
The entity should not be able to acquire resources outside the lifetime of the config entry. If the config entry is removed all connected entities should also be removed.

Reply:
You are correct, I misspoke.

The `DlnaDmrEntity` can acquire and release resources on its own, *inside* the lifetime of the config entry, but more often than just the setup and removal of the config entry. Specifically, each time it connects to the underlying device it will acquire certain resources, and each time it disconnects it should release those resources. Connection and disconnection can happen multiple times within the lifetime of the config entry, but when the config entry is removed then the device should be disconnected (if it was connected at the time) and all resources released.

Reply:
Right. And what is part of this test tests something that is not tested in `test_resource_lifecycle`?

Reply:
Looks like bith the ssdp callbacks and the polling causes connection and disconnection events. Ok.